src/libs/swapAndBridge/swapAndBridge.ts

```diff
@@ -463,6 +474,89 @@ const getSlippage = (
     : (delimeter / Math.ceil(Number(fromAmountInUsd) / 20000)).toPrecision(2)
 }
 
+export const calculateAmountWarnings = (
+  selectedRoute: SwapAndBridgeQuote['selectedRoute'],
+  fromAmountInFiat: string,
+  fromAmount: string,
+  fromSelectedTokenDecimals: number
+):
+  | { type: 'highPriceImpact'; percentageDiff: number }
+  | {
+      type: 'slippageImpact'
+      possibleSlippage: number
+      minInUsd: number
+      minInToken: string
+      symbol: string
+    }
+  | null => {
+  if (!selectedRoute) return null
+
+  let inputValueInUsd = 0
+  const outputValueInUsd = selectedRoute.outputValueInUsd
+
+  try {
+    inputValueInUsd = Number(fromAmountInFiat)
+  } catch (error) {
+    // silent fail
+  }
+  if (!inputValueInUsd) return null
+
+  try {
+    const sanitizedFromAmount = getSanitizedAmount(fromAmount, fromSelectedTokenDecimals)
+
+    const bigintFromAmount = parseUnits(sanitizedFromAmount, fromSelectedTokenDecimals)
+
+    if (bigintFromAmount !== BigInt(selectedRoute.fromAmount)) return null
+
+    // Can be negative if the output is higher
+    // (possible during arbitrage swaps)
+    const difference = inputValueInUsd - outputValueInUsd
+
+    const percentageDiff = (difference / inputValueInUsd) * 100
+
+    if (percentageDiff >= 5) {
+      return {
+        type: 'highPriceImpact',
+        percentageDiff
+      }
+    }
+
+    // try to calculate the slippage
+    const minAmountOutInWei = BigInt(
+      selectedRoute.userTxs[selectedRoute.userTxs.length - 1].minAmountOut
+    )
+    const minInUsd = safeTokenAmountAndNumberMultiplication(
+      minAmountOutInWei,
+      selectedRoute.toToken.decimals,
+      Number(selectedRoute.toToken.priceUSD)
+    )
+    const allowedSlippage =
+      Number(inputValueInUsd) < 400
+        ? 1.05
+        : Number((0.005 / Math.ceil(Number(inputValueInUsd) / 20000)).toPrecision(2)) * 100 + 0.01
+    const possibleSlippage = (1 - Number(minInUsd) / outputValueInUsd) * 100
+    // @precautionary if
+    const diffBetweenQuoteAndMinAmount =
+      outputValueInUsd > Number(minInUsd) ? outputValueInUsd - Number(minInUsd) : 0
+
+    // It seems a bit odd to display a slippage warning only if the difference
+    // is > $50?
```

Comment:
I think the reasoning behind this was to bypass the check when 1) Swapping small amounts and 2) Then the $ diff is "whatever". Not sure if this is optimal (checking **difference** of $50), please cross-check with @Oxbobby and @Ivshti , but that's now a show-stopped for this PR since it was like this before your changes anyways.